dunwoody_disaster/CharacterFactory.py

```diff
@@ -119,6 +122,15 @@ def reset(self):
         self.set_magic(self.maxMagic)
         self.set_stamina(self.maxStamina)
 
+    def reload(self):
+        """Reloads the character's health, stamina, and magic."""
+        char = CharacterFactory.LoadCharacter(self.name)
+        self = char
+        self.set_health(char.curHealth)
+        self.set_magic(char.curMagic)
+        self.set_stamina(char.curStamina)
+        print(f"loading{self.name} health: {self.curHealth}")
```

Comment:
I think `self` here is a local variable, so any changes made to it won't be affected outside of this scope.
```suggestion
    def reload(self):
        """Reloads the character's health, stamina, and magic from the last save point."""
        char = CharacterFactory.LoadCharacter(self.name)
        self.set_health(char.curHealth)
        self.set_magic(char.curMagic)
        self.set_stamina(char.curStamina)
```

Reply:
Yeah i think that was part of the issues I was having